src/client/Client.cc

```diff
@@ -8259,7 +8258,11 @@ int Client::lookup_ino(inodeno_t ino, const UserPerm& perms, Inode **inode)
   return r;
 }
 
-
+int Client::lookup_ino_locked(inodeno_t ino, const UserPerm& perms, Inode **inode)
+{
+  Mutex::Locker lock(client_lock);
+  return lookup_ino(ino, perms, inode);
+}
```

Comment:
This approach looks much better.

My only complaint is that the *_locked suffix usually indicates the function that should be called when you already hold the lock. In other words, lookup_ino and lookup_ino_locked should have their names swapped here. Same for the other functions: swap the names of lookup_parent and lookup_parent_locked.

Reply:
maybe use lookup_ino_unlock/lookup_parent_unlock/ ll_forget_unlock/lookup_name_unlock?
"locked" said it already holds the lock.

Reply:
To be consistent with Client.cc, perhaps we should  use _lookup_ino to means the locked version.  lookup_ino can take the lock and then call _lookup_ino. please take reference as "_map_faked_ino" and "map_faked_ino"

Reply:
@taodd 
Good advice, but will change the current API interface name, so I prefer the current way